src/api/xccl.h

```diff
@@ -435,17 +436,27 @@ typedef struct xccl_coll_algorithm {
     uint8_t id : 7;
 } xccl_coll_algorithm_t;
 
+typedef struct xccl_stream {
+    ucs_memory_type_t mem_type;
```

Comment:
Why "mem_type" is part of stream ? Is it an identifier of stream "type", like CUDA_STREAM, ROCM_STREAM ... ? Maybe need to define XCCL_STREAM_TYPE?